examples/rdl.py

```diff
@@ -0,0 +1,710 @@
+args.add_argument("--channels", type=int, default=128)
+args.add_argument("--out_channels", type=int, default=1)
+args.add_argument("--aggr", type=str, default="sum")
+args.add_argument("--norm", type=str, default="batch_norm")
+
+# Training parameters
+args.add_argument("--epochs", type=int, default=10)
+args.add_argument("--learning_rate", type=float, default=0.005)
+
+args = args.parse_args()
+
+
+class GloveTextEmbedding:
+    """GloveTextEmbedding based on SentenceTransformer."""
+    def __init__(self, device: Optional[torch.device] = None):
+        self.model = SentenceTransformer(
+            "sentence-transformers/average_word_embeddings_glove.6B.300d",
+            device=device,
+        )
+
+    def __call__(self, sentences: List[str]) -> Tensor:
+        return torch.from_numpy(self.model.encode(sentences))
+
+
+class HeteroEncoder(torch.nn.Module):
+    r"""HeteroEncoder based on PyTorch Frame.
+
+    A heterogeneous encoder that processes different node types using PyTorch
+    Frame models. For each node type, it creates a separate encoder model
+    that processes the node features according to their data types
+    (categorical, numerical, etc).
+
+    Args:
+        channels (int): The output channels for each node type.
+        node_to_col_names_dict
+            (Dict[NodeType, Dict[torch_frame.stype, List[str]]]):
+            A dictionary mapping from node type to column names dictionary
+            compatible to PyTorch Frame.
+        node_to_col_stats (Dict[NodeType, Dict[str, Dict[StatType, Any]]]):
+            A dictionary containing statistics for each column
+            in each node type. Used for feature normalization and encoding.
+        torch_frame_model_cls: Model class for PyTorch Frame. The class object
+            takes :class:`TensorFrame` object as input and outputs
+            :obj:`channels`-dimensional embeddings. Default to
+            :class:`torch_frame.nn.ResNet`.
+        torch_frame_model_kwargs (Dict[str, Any]): Keyword arguments for
+            :class:`torch_frame_model_cls` class. Default keyword argument is
+            set specific for :class:`torch_frame.nn.ResNet`. Expect it to
+            be changed for different :class:`torch_frame_model_cls`.
+        default_stype_encoder_cls_kwargs (Dict[torch_frame.stype, Any]):
+            A dictionary mapping from :obj:`torch_frame.stype` object into a
+            tuple specifying :class:`torch_frame.nn.StypeEncoder` class and its
+            keyword arguments :obj:`kwargs`.
+    """
+    def __init__(
+        self,
+        channels: int,
+        node_to_col_names_dict: Dict[NodeType, Dict[torch_frame.stype,
+                                                    List[str]]],
+        node_to_col_stats: Dict[NodeType, Dict[str, Dict[StatType, Any]]],
+        torch_frame_model_cls=ResNet,
+        torch_frame_model_kwargs: Dict[str, Any] = {
+            "channels": 128,
+            "num_layers": 4,
+        },
```

Comment:
Usually, we don't want to instantiate mutable objects as part of init. It's probably okay for an example though.